types/react-select/lib/styles.d.ts

```diff
@@ -0,0 +1,99 @@
+import {
+  containerCSS,
+  indicatorsContainerCSS,
+  valueContainerCSS,
+} from './components/containers';
+import { css as controlCSS } from './components/Control';
+import { groupCSS, groupHeadingCSS } from './components/Group';
+import {
+  clearIndicatorCSS,
+  dropdownIndicatorCSS,
+  loadingIndicatorCSS,
+  indicatorSeparatorCSS,
+} from './components/indicators';
+import { inputCSS } from './components/Input';
+import { placeholderCSS } from './components/Placeholder';
+import { optionCSS } from './components/Option';
+import {
+  menuCSS,
+  menuListCSS,
+  menuPortalCSS,
+  noOptionsMessageCSS,
+  loadingMessageCSS,
+} from './components/Menu';
+import { css as singleValueCSS } from './components/SingleValue';
+import {
+  multiValueCSS,
+  multiValueLabelCSS,
+  multiValueRemoveCSS,
+} from './components/MultiValue';
+
+export type Props = { [key: string]: any };
+
+/**
+ * @param base -- the component's default style
+ * @param state -- the component's current state e.g. `isFocused`
+ * @returns
+ */
+export type styleFn = (base: any, state: any) => any;
```

Comment:
I know it's the same as in original Flow typings, but those `any` are demotivating...

Actually, aren't `base` and return type just `React.CSSProperties`?

Reply:
Again thanks for the pointer, @burtek! :)

Yes, that function (and several others) should be returning css properties. I'll fix this.

Reply:
It would also be nice to type `state`, although this might be hard to achieve, as there is literally no docs on that

Reply:
`state` appears to be the props for the component that is being styled. For example,

    export type Styles<OptionType> = {
        ...
        option?: (base: CSSProperties, state: OptionProps<OptionType>) => CSSProperties,
        ...
    };

I actually did away with the `StylesConfig` in my definitions, because it seemed redundant.